nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/ExcelUtils.java

```diff
@@ -16,15 +16,45 @@
  */
 package org.apache.nifi.excel;
 
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.DateUtil;
 import org.apache.poi.ss.usermodel.Row;
 
+import java.util.Locale;
+
 public class ExcelUtils {
     static final String FIELD_NAME_PREFIX = "column_";
+    private static final DataFormatter DATA_FORMATTER = new DataFormatter(Locale.getDefault());
+
+    static {
+        DATA_FORMATTER.setUseCachedValuesForFormulaCells(true);
+    }
 
     private ExcelUtils() {
     }
 
     public static boolean hasCells(final Row row) {
         return row != null && row.getFirstCellNum() != -1;
     }
+
+    public static String getFormattedCellValue(Cell cell) {
+        if (cell != null) {
+            /* DataFormatter should be used in most cases unless an exception will be thrown or
+              string localization may occur which could result in later exceptions. A cell which has a formula cannot
+              format a cell which has an error value and an exception will be thrown. In addition, a cell whose type
+              is numeric when formatted will take the current locale into consideration which could prevent later parsing
+              as a float/double.*/
+            if (cell.getCellType().equals(CellType.FORMULA) && cell.getCachedFormulaResultType().equals(CellType.ERROR)) {
+                return cell.getStringCellValue();
+            } else if (cell.getCellType().equals(CellType.NUMERIC) && !DateUtil.isCellDateFormatted(cell)) {
```

Comment:
It looks like it would be helpful to declare a variable for the `cellType`, making these conditionals a bit more readable than calling `getCellType()` in both blocks.